src/components/Calendar.js

```diff
@@ -0,0 +1,225 @@
+const React = require('react');
+const Radium = require('radium');
+const moment = require('moment');
+
+const Icon = require('./Icon');
+
+const StyleConstants = require('../constants/Style');
+
+const Calendar = React.createClass({
+  propTypes: {
+    locale: React.PropTypes.string,
+    minimumDate: React.PropTypes.number,
+    onDateSelect: React.PropTypes.func,
+    primaryColor: React.PropTypes.string,
+    selectedDate: React.PropTypes.number,
+    style: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      locale: 'en',
+      onDateSelect () {},
+      primaryColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      currentDate: this.props.selectedDate || moment().unix()
+    };
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.selectedDate && newProps.selectedDate !== this.props.selectedDate) {
```

Comment:
Copied the code over from the DatePicker component so I can't say for sure but it appears that whoever wrote it was trying to protect against undefined errors.